docs/reference/search/search-your-data/search-using-query-rules.asciidoc

```diff
@@ -0,0 +1,150 @@
+--
+* The criteria for which this rule will match. Criteria is defined as `type`, `metadata`, and `values`. The allowed types are:
++
+--
+* `exact` will match when the rule metadata matches the specified value exactly.
+* `fuzzy` will match when the rule metadata matches the specified value within an allowed {wikipedia}/Levenshtein_distance[Levenshtein edit distance].
+* `prefix` will match when the rule metadata starts with the specified value.
+* `suffix` will match when the rule metadata ends with the specified value.
+* `contains` will match when the rule metadata contains the specified value.
+* `lt` will match when the rule metadata is less than the specified value.
+* `lte` will match when the rule metadata is less than or equal to the specified value.
+* `gt` will match when the rule metadata is greater than the specified value.
+* `gte` will match when the rule metadata is greater than or equal to the specified value.
+* `always` will always match for all rule queries.
+--
+* The actions to take when the rule matches a query:
++
+--
+* `ids` will pin the specified <<mapping-id-field,`_id`>>s.
+* `docs` will pin the specified documents in the specified indices.
+--
+
+[discrete]
+[[add-query-rules]]
+=== Add query rules
+
+You can add query rules using the <<put-query-ruleset>> call. This adds a ruleset containing one or more query rules that will be applied to queries that match their specified criteria.
+
+The following command will create a query ruleset called `my-ruleset` with two pinned document rules:
+
+* The first rule will generate a <<query-dsl-pinned-query>> pinning the <<mapping-id-field,`_id`>>s `id1` and `id2` when the `user.query` matches the value `pugs`.
+* The second rule will generate a <<query-dsl-pinned-query>> pinning the <<mapping-id-field, `_id`>> of `id3` specifically from the `my-index-000001` index and `id4` from the `my-index-000002` index when the `user.query` matches the value `beagles`.
+
+[source,console]
+----
+PUT /_query_rules/my-ruleset
+{
+  "rules": [
+    {
+      "rule_id": "rule1",
+      "type": "pinned",
+      "criteria": [
+        {
+          "type": "exact",
+          "metadata": "user.query",
+          "values": [ "pugs" ]
+        }
+      ],
+      "actions": {
+        "ids": [
+          "id1",
+          "id2"
+        ]
+      }
+    },
+    {
+      "rule_id": "rule2",
+      "type": "pinned",
+      "criteria": [
+        {
+          "type": "exact",
+          "metadata": "user.query",
+          "values": [ "beagles" ]
+        }
+      ],
+      "actions": {
+        "docs": [
+          {
+            "_index": "my-index-000001",
+            "_id": "id3"
+          },
+          {
+            "_index": "my-index-000002",
+            "_id": "id4"
+          }
+        ]
+      }
+    }
+  ]
+}
+----
+// TEST[skip:TBD]
```

Comment:
I guess tests will be added in a subsequent PR

Reply:
Good callout. Yes, I will plan a followup PR for tests.